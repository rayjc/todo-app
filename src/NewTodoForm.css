.NewTodoForm-label {
    margin: auto 5px;
}