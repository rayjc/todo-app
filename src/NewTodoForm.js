import React, { useState } from 'react';
import './NewTodoForm.css'

const NewTodoForm = ({ addTodo }) => {
  const [todo, setTodo] = useState("");

  const handleSubmit = (e) => {
    e.preventDefault();
    // console.log("Add ", todo);
    if (todo.length) {
      // add todo
      addTodo(todo);

      setTodo("");
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="todo-input" className="NewTodoForm-label">Task:</label>
      <input
        type="text"
        id="todo-input"
        name="todo"
        value={todo}
        onChange={(e) => setTodo(e.target.value)}
      />
      <button type="submit">Add</button>
    </form>
  )
}


export default NewTodoForm