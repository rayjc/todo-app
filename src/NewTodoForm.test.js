import React from "react";
import { render } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

it("renders without crashing", function() {
  render(<NewTodoForm />);
});

it("should match default snapshot", function() {
  const { asFragment } = render(
    <NewTodoForm addTodo={jest.fn()} />
  );
  expect(asFragment()).toMatchSnapshot();
})