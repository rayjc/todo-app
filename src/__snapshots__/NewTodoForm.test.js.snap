// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match default snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      class="NewTodoForm-label"
      for="todo-input"
    >
      Task:
    </label>
    <input
      id="todo-input"
      name="todo"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add
    </button>
  </form>
</DocumentFragment>
`;
